src/style/attributes.rs

```diff
@@ -0,0 +1,111 @@
+use crate::style::Attribute;
+use std::ops::{BitAnd, BitOr, BitXor};
+
+/// a bitset for all possible attributes
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq)]
+pub struct Attributes(u64);
```

Comment:
Do we really need a u64 for this? We can go way smaller right? Even something like a u8 would already be enough. Or am I mistaken?

Reply:
There are 24 values. So a u8 can't store them all.
A u32 would be enough but then we need another implementation for `Attribute.bytes()`.
A possibility would be to remove the SGR indexes from the enum and hardcode them in the unix implementation of SetAttribute and SetAttributes. Which would probably not be dirtier.